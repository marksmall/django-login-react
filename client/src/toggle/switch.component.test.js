import React from 'react';
import { shallow } from 'enzyme';

import Switch from './switch.component';

describe('Switch Component', () => {
  it('should be in the on style', () => {
    const on = true;
    const onClick = jest.fn();

    const testee = shallow(<Switch on={on} onClick={onClick} />);

    expect(testee).toMatchSnapshot();
    expect(testee.find('.toggle-btn-on').length).toEqual(1);
    expect(testee.find('.toggle-btn-off').length).toEqual(0);
  });

  it('should be in the off style', () => {
    const on = false;
    const onClick = jest.fn();

    const testee = shallow(<Switch on={on} onClick={onClick} />);

    expect(testee.find('.toggle-btn-on').length).toEqual(0);
    expect(testee.find('.toggle-btn-off').length).toEqual(1);
  });

  it('should switch from off to on style', () => {
    const on = true;
    const onClick = jest.fn();

    const testee = shallow(<Switch on={on} onClick={onClick} />);

    expect(testee.find('.toggle-btn-on').length).toEqual(1);
    expect(onClick.mock.calls.length).toEqual(0);

    testee.find('button').simulate('click');
    expect(onClick.mock.calls.length).toEqual(1);
  });
});
