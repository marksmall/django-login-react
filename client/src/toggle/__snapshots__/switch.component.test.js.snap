// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Component should be in the on style 1`] = `
<div>
  <input
    checked={true}
    className="toggle-input"
    onChange={[Function]}
    type="checkbox"
  />
  <button
    aria-label="Toggle"
    className="toggle-btn
      toggle-btn-on"
    disabled={false}
    onClick={[MockFunction]}
  />
</div>
`;
